vertical-pod-autoscaler/pkg/recommender/routines/recommender.go

```diff
@@ -35,7 +37,6 @@ import (
 
 var (
 	checkpointsWriteTimeout = flag.Duration("checkpoints-timeout", time.Minute, `Timeout for writing checkpoints since the start of the recommender's main loop`)
-	minCheckpointsPerRun    = flag.Int("min-checkpoints", 10, "Minimum number of checkpoints to write per recommender's main loop")
```

Comment:
Just a thought, Kubernetes doesn't remove flags in a single release (or ever?), they are normally deprecated over time.

What if this flag was still active, but spat out a warning if used? May be its description also updated:
```
minCheckpointsPerRun    = flag.Int("min-checkpoints", 10, "Minimum number of checkpoints to write per recommender's main loop. DEPRECATED FLAG and will be removed in the future.")
```

Then in a later release it can be remove altogether?

Reply:
And by "active" you mean the recommender wouldn't crash if you specify it, but its contents are ignored? That's probably a good idea – not sure if people are reading their logs more thoroughly than the release notes, though ;) I understand this is a pattern in k8s that we could (should?) follow as well.

However, if possible, I'd rather not like to keep it actually working, given that we would be changing twice how the minimum amount of checkpoints written works and how  much more complex the code is compared to just removing this entirely.

Do you want me to keep the flag but make it a no-op, such that people can see a deprecation warning in their logs?

Reply:
> And by "active" you mean the recommender wouldn't crash if you specify it, but its contents are ignored?

Correct. 

> That's probably a good idea – not sure if people are reading their logs more thoroughly than the release notes, though ;) I understand this is a pattern in k8s that we could (should?) follow as well.

Agreed. Keeping the flag around (without it doing anything) for a few releases isn't that much better than disabling it outright, but in theory we could check if 3rd party helm charts have updated to stop supporting this flag.
 
> Do you want me to keep the flag but make it a no-op, such that people can see a deprecation warning in their logs?

Correct, my vote is that the flag still exists, but is a no-op, and spits out a deprecation warning to users.

Reply:
done.